packages/babel-parser/src/parser/statement.js

```diff
@@ -1757,18 +1757,9 @@ export default class StatementParser extends ExpressionParser {
 
   isAsyncFunction(): boolean {
     if (!this.isContextual("async")) return false;
-
-    const { pos } = this.state;
-
-    skipWhiteSpace.lastIndex = pos;
-    const skip = skipWhiteSpace.exec(this.input);
-
-    if (!skip || !skip.length) return false;
-
-    const next = pos + skip[0].length;
-
+    const next = this.nextTokenStart();
     return (
-      !lineBreak.test(this.input.slice(pos, next)) &&
+      !lineBreak.test(this.input.slice(this.state.pos, next)) &&
```

Comment:
@JLHwung You don't gain performance here. Try replace `!lineBreak.test(this.input.slice(this.state.pos, next))` with this

```ts
const { input, pos } = this;
const nextChar = input.slice(pos, next);
return ( (nextChar=== 0x10 || nextChar === 0x13 || (nextChar^ 0x2028) <= 1) &&
input.slice(next, next + 8) === "function" && (next + 8 === this.length ||
        !isIdentifierChar(input.charCodeAt(next + 8)))
```

Not ideal either, but an improvement :)

Reply:
`isIdentifierChar` can be super optimized, but only use a direct lookup for it. Similiar to what I did [here](https://github.com/meriyah/meriyah/blob/master/src/lexer/identifier.ts#L15). But I recently found a faster way to scan for identifier than what V8 does which include direct table lookup without bitmasks etc.
 
Btw. I couldn't get babel parser to run in my benchmark, but where do I find a benchmark with it online? And try run [this benchmark](https://meriyah.github.io/meriyah/performance/) and see how Meriyah does it vs Acorn.

Reply:
The source for the benchmark is [located here](https://github.com/meriyah/meriyah/tree/gh-pages/performance) and the website is in the root folder.

Estimate 14 days hard work and you could replicate the Babel parser if you write it from scratch.  Another 6 - 8 days to get all plug-ins working. And the Babel parser should perform same as Meriyah :)

Reply:
@JLHwung Meriyah's REPL is [located here](https://meriyah.github.io/meriyah/) in case of interest. Inspired by Babel's REPL because I found out that the REPL was loading the page very slow too :)

Reply:
@JLHwung You mentioned table lookup table for identifier scanning. I would say that the V8 solution isn't as fast as it could be either, but what you can do is have a table lookup for the token kinds. Then you know that keywords are only lower letters, and e.g. no keyword starts with letter 'u'. This knowledge you can use to optimize for this, and you shouldn't need to check what's between "k" and "d" in "keyword" if you use a while loop iteration and checks for IdentifierPart chars only. That way you would only need to check what's follow after "keyword". That's faster than what V8 does. And you skip unnecessary branching. 
I just implemented this in my own lexer refactoring - seen [here](https://github.com/meriyah/meriyah/blob/recovery/src/scanner/identifier.ts#L9)

I just mentioned it because you mentioned it first, and it could be a good optimization trick for Babel :)